hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/BackgroundService.java

```diff
@@ -103,8 +105,25 @@ public void runPeriodicalTaskNow() throws Exception {
   }
 
   // start service
-  public void start() {
-    exec.scheduleWithFixedDelay(service, 0, interval, unit);
+  public synchronized void start() {
+    if (scheduledHandle != null && !scheduledHandle.isCancelled()) {
+      LOG.warn("Background service {} is already running", serviceName);
+      return;
+    }
+    scheduledHandle = exec.scheduleWithFixedDelay(service, 0, interval, unit);
+  }
+
+  protected synchronized void setInterval(long newInterval, TimeUnit newUnit) {
+    this.interval = newInterval;
+    this.unit = newUnit;
+  }
+
+  public synchronized void stop() {
+    LOG.info("Stopping {}", serviceName);
+    if (scheduledHandle != null) {
+      scheduledHandle.cancel(false); // don't interrupt running tasks
```

Comment:
We may need interrupt running task make wait till its completed. this is as when start() new task, should not conflict is already running task. DirectoryDeletingTask can have reverse impact.